src/System.Numerics.Tensors/System/Numerics/TensorView.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Numerics
+{
+    /// <summary>
+    /// Provides a view into, or slice of, a backing Tensor object. TensorView is used to provide
+    /// a projection of a subset of data from the backing Tensor.
+    /// </summary>
+    internal class TensorView<T> : Tensor<T>
+    {
+        private readonly Tensor<T> _backingTensor;
+        private readonly int[] _stridesIntoBackingTensor;
+        private readonly int _offset;
+
+        internal TensorView(Tensor<T> backingTensor, ReadOnlySpan<int> dimensions, ReadOnlySpan<int> stridesIntoBackingTensor, int offset)
+            : base(dimensions, backingTensor.IsReversedStride)
+        {
+            if (stridesIntoBackingTensor.Length == 0)
+            {
+                throw new ArgumentException("Strides must contain elements.", nameof(stridesIntoBackingTensor));
+            }
+            if (dimensions.Length != stridesIntoBackingTensor.Length)
+            {
+                throw new ArgumentException($"dimensions.Length '{dimensions.Length}' must match backingTensorStrides.Length '{stridesIntoBackingTensor.Length}'");
+            }
+
+            _backingTensor = backingTensor;
+            _offset = offset;
+            _stridesIntoBackingTensor = new int[stridesIntoBackingTensor.Length];
+            for (int i = 0; i < stridesIntoBackingTensor.Length; i++)
+            {
+                if (stridesIntoBackingTensor[i] < 1)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(stridesIntoBackingTensor), "Strides must be positive and non-zero");
+                }
+                _stridesIntoBackingTensor[i] = stridesIntoBackingTensor[i];
+            }
+        }
+
+        public override T this[ReadOnlySpan<int> indices]
+        {
+            get
+            {
+                EnsureValidIndices(indices);
+
+                return _backingTensor.GetValue(_offset + ArrayUtilities.GetIndex(_stridesIntoBackingTensor, indices));
+            }
+            set
+            {
+                EnsureValidIndices(indices);
+
+                _backingTensor.SetValue(_offset + ArrayUtilities.GetIndex(_stridesIntoBackingTensor, indices), value);
+            }
+        }
+
+        public override T GetValue(int index)
+        {
+            EnsureValidIndex(index);
+
+            Span<int> indices = Rank < ArrayUtilities.StackallocMax ? stackalloc int[Rank] : new Span<int>(new int[Rank]);
```

Comment:
use implicit cast here and elsewhere

Reply:
@ahsonkhan, your input here is misleading. What core of what processor in what NUMA domain will the code in question execute on? I have not even looked at the code which provides the logic associated with the property referenced and I will ask how will such property as `ArrayUtilities.StackallocMax` help if the stack size has been intentionally modified above or below auch limits as provided by the property in question... yet again and as always we thank you for your input..

Reply:
I don't understand what you mean. See the following for what I am suggesting: https://github.com/dotnet/corefxlab/pull/1927#discussion_r150354779

Reply:
Obviously or maybe only to me, it seems you do not understand the locality of the execution of the code which you are suggesting be utilized for some improvement when in reality it may actually not be an improvement and furthermore may introduce bugs; finally - for what I see at the link or lack thereof it will not chnage the nature of my statements which remain true.